core/includes/menu.inc

```diff
@@ -1837,6 +1837,29 @@ function theme_menu_local_actions($variables) {
 }
 
 /**
+ * Returns rendered HTML for a menu toggle.
+ *
+ * @param $variables
+ *   An associative array containing:
+ *     - #enabled: A boolean indicating whether the menu toggle is active.
+ *     - hash: Unique string id to match the label to the input.
+ *
+ * @see theme_menu_toggle()
+ * @ingroup themeable
+ */
+function theme_menu_toggle($variables) {
+  $output = '';
+  if ($variables['enabled']) {
+    $output  = '<input id="menu-toggle-state-' . $variables['hash'] . '" class="menu-toggle-state element-invisible" type="checkbox"  aria-controls="menu-toggle-state-' . $variables['hash'] . '" />';
+    $output .= '<label class="menu-toggle-button" for="menu-toggle-state-' . $variables['hash'] . '">';
+    $output .= '  <span class="menu-toggle-button-icon"></span><span class="menu-toggle-button-text">' . t($variables['text']) . '</span>';
```

Comment:
The text should be translated before it gets passed in. See system_menu_block_build().